kubernetes/kubernetes.libsonnet

```diff
@@ -36,6 +36,34 @@ local objectMetadata(object, config) =
     } + config.podAnnotations
   );
 
+// Function to create volumes in deployment
+local volumeService(volumeSpec) = [
+  local volume = k.core.v1.volume;
+  local name = volume.withName(vol.name);
+  volume.withName(vol.name)
+  + (if std.objectHas(vol, 'awsElasticBlockStore') then volume.awsElasticBlockStore.withFsType(vol.awsElasticBlockStore.fsType) else {})
+  + (if std.objectHas(vol, 'awsElasticBlockStore') then volume.awsElasticBlockStore.withVolumeID(vol.awsElasticBlockStore.volumeID) else {})
+  + (if std.objectHas(vol, 'awsElasticBlockStore') then volume.awsElasticBlockStore.withPartition(vol.awsElasticBlockStore.partition) else {})
+
+  + (if std.objectHas(vol, 'configMap') then volume.fromConfigMap(name, configMapName=vol.configMap.name) else {})
+
+  + (if std.objectHas(vol, 'secret') then volume.fromSecret(name, secretName=vol.secret.name) else {})
+
+  
+  for vol in volumeSpec
+];
+
+// Function to create volumeMounts in container
+local containerVolume(volumeSpec) = [
+  local volume = k.core.v1.volumeMount;
+  volume.withMountPath(vol.mountPath)
+  // + volume.withMountPropagation(volumeSpec.mountPropagation)
+  + volume.withName(vol.name)
+  // + volume.withReadOnly(volumeSpec.readOnly)
+  + volume.withSubPath(vol.subPath)
```

Comment:
you'll probably want to hide this behind a condition. We don't use subpaths for all volumes.